server/settings_test.go

```diff
@@ -192,6 +196,56 @@ func TestSettings(t *testing.T) {
 				assert.True(t, resg.Payload.Settings.SttEnabled)
 			})
 
+			//Verify Failed checks alerts count in alertmanager
+			t.Run("VerifyFailedChecksInAlertmanager", func(t *testing.T) {
+				defer restoreDefaults(t)
+				// Enabling STT
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+
+				activeAlerts := true
+				silencedAlerts := false
+				alertsCount := 0
+
+				//120 sec ping for failed checks alerts to appear in alertmanager
```

Comment:
Will it always work correctly? Even if there is no pmm-agent connected? I think it's better to expect exact alert instead of random one.

Reply:
Well the alert messages are continuously updated and it is hard to maintain such test. That is why i decided to verify the structure of alerts